libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -496,6 +500,18 @@ export class CartService {
       );
     }
 
+    if (cart.eligibilityStatus === CartEligibilityStatus.INVALID) {
+      return {
+        ...cart,
+        state: CartState.FAIL,
+        errorReasonId: CartErrorReasonId.CartEligibilityStatusInvalid,
```

Comment:
If the eligibility is INVALID, I'd expect the cart to be set to error state.

A cart in error state would have `cart.state: CartState.FAIL` and `errorReasonId: CartErrorReasonId.xxx`

This current change only returns valid values for `state` and `errorReasonId` but doesn't update the underlying Cart DB record. Since our goal with SP3 is for the frontend to be a reflection of the Cart, its important for Cart in the DB to match what the frontend is showing.